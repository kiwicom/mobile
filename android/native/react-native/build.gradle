buildscript {
    apply from: "../../dependencies.gradle"

    repositories {
        google()
        jcenter()
        mavenCentral()

        maven {
            url "https://plugins.gradle.org/m2/"
        }
    }
    dependencies {
        classpath "com.android.tools.build:gradle:$gradlePluginVersion"
        // Needed to assemble react-native
        classpath 'de.undercouch:gradle-download-task:3.4.3'
    }
}

subprojects {
    apply plugin: 'maven'
    apply plugin: 'com.android.library'

    repositories {
        google()
        jcenter()

        maven {
            url "$rootDir/../../node_modules/react-native/android"
        }
        maven {
            url 'http://trinerdis.cz:8000/repository/snapshots/'
        }
    }

    buildDir = "$rootProject.projectDir/build"

    tasks.create('uploadTrinerdis', Upload.class) {
        configuration = project.configurations.archives
        repositories {
            mavenDeployer {
                repository(url: "http://trinerdis.cz:8000/repository/snapshots/") {
                    authentication(userName: "deployment", password: System.getenv("ANDROID_DEPLOYMENT_PASSWORD"))
                    pom.version = "$version"
                    pom.artifactId = "react-native"
                    pom.groupId = "com.facebook.react"
                }
            }
        }
    }

}

task clean(type: Delete) {
    delete rootProject.buildDir
}
