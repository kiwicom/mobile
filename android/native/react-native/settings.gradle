include ':react-native'
project(':react-native').projectDir = new File(
        rootProject.projectDir, '../../../node_modules/react-native/ReactAndroid'
)
