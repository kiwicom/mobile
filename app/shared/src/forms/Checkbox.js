// @flow

import * as React from 'react';
import {
  View,
  StyleSheet,
  TouchableWithoutFeedback,
  Platform,
} from 'react-native';

import Color from '../Color';
import Icon from '../Icon';
import type { StylePropType } from '../../types/Styles';

type Props = {|
  onPress: () => void,
  children: React.Node,
  isChecked?: boolean,
  style?: StylePropType,
|};

export default function Checkbox(props: Props) {
  return (
    <TouchableWithoutFeedback onPress={props.onPress}>
      <View style={[styles.wrapper, props.style]}>
        <View style={styles.label}>{props.children}</View>
        {props.isChecked && (
          <View style={styles.check}>
            <Icon
              name="check"
              size={Platform.select({
                android: 25, // solves checkbox row jumping on Android devices
                ios: 26,
              })}
              color={Color.brand}
            />
          </View>
        )}
      </View>
    </TouchableWithoutFeedback>
  );
}

const styles = StyleSheet.create({
  wrapper: {
    flexDirection: 'row',
    paddingTop: 8,
    paddingBottom: 8,
  },
  label: {
    flex: 3,
    justifyContent: 'center',
    height: 26,
  },
  check: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'flex-end',
  },
});
