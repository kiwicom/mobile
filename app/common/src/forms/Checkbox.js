// @flow

import * as React from 'react';
import { View, StyleSheet, TouchableWithoutFeedback } from 'react-native';

import Color from '../Color';
import Icon from '../Icon';

type Props = {|
  onPress: () => void,
  children: React.Node,
  isChecked?: boolean,
  // $FlowFixMeProps
  style?: Object,
|};

export default function Checkbox(props: Props) {
  return (
    <TouchableWithoutFeedback onPress={props.onPress}>
      <View style={[styles.wrapper, props.style]}>
        <View style={styles.label}>{props.children}</View>
        {props.isChecked && (
          <View style={styles.check}>
            <Icon name="check" size={26} color={Color.brand} />
          </View>
        )}
      </View>
    </TouchableWithoutFeedback>
  );
}

const styles = StyleSheet.create({
  wrapper: {
    flexDirection: 'row',
    paddingTop: 8,
    paddingBottom: 8,
  },
  label: {
    flex: 3,
    justifyContent: 'center',
    height: 26,
  },
  check: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'flex-end',
  },
});
