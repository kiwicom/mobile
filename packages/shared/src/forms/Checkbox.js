// @flow

import * as React from 'react';
import { View, Platform } from 'react-native';
import { defaultTokens } from '@kiwicom/mobile-orbit';

import TouchableWithoutFeedback from '../TouchableWithoutFeedback';
import StyleSheet from '../PlatformStyleSheet';
import type { StylePropType } from '../../types/Styles';
import TextIcon from '../icons/TextIcon';

type Props = {|
  +onPress: () => void,
  +children: React.Node,
  +isChecked?: boolean,
  +style?: StylePropType,
|};

export default function Checkbox(props: Props) {
  const icon = (
    <TextIcon
      code={Platform.OS === 'ios' ? 'S' : '\ue013'}
      style={styles.iconStyle}
    />
  );

  return (
    <TouchableWithoutFeedback onPress={props.onPress}>
      <View style={[styles.wrapper, props.style]}>
        <View style={styles.label}>{props.children}</View>
        <View style={styles.checkWrapper}>
          {props.isChecked ? icon : <View style={styles.notChecked} />}
        </View>
      </View>
    </TouchableWithoutFeedback>
  );
}

const styles = StyleSheet.create({
  wrapper: {
    flexDirection: 'row',
    paddingTop: 8,
    paddingBottom: 8,
  },
  label: {
    flex: 3,
    justifyContent: 'center',
    height: 26,
  },
  checkWrapper: {
    justifyContent: 'center',
    alignItems: 'flex-end',
  },
  iconStyle: {
    color: defaultTokens.paletteProductNormal,
    ios: {
      width: 26,
      height: 26,
      fontSize: 28,
    },
    android: {
      width: 24,
      height: 24,
      borderWidth: 1,
      borderColor: defaultTokens.paletteProductNormal,
      borderRadius: 100,
      fontSize: 24,
    },
  },
  notChecked: {
    width: 24,
    height: 24,
    borderRadius: 100,
    borderWidth: 1,
    borderColor: defaultTokens.paletteInkLighter,
  },
});
