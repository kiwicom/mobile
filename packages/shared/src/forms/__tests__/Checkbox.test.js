// @flow strict

import * as React from 'react';
import { Platform } from 'react-native';
import { PlaygroundRenderer } from '@kiwicom/mobile-playground';
import { Translation } from '@kiwicom/mobile-localization';
import { defaultTokens } from '@kiwicom/mobile-orbit';
import StyleSheet from '@kiwicom/mobile-shared/src/PlatformStyleSheet';

import Checkbox from '../Checkbox';

const noop = () => {};

it('not checked', () => {
  PlaygroundRenderer.render(
    <Checkbox onPress={noop} isChecked={false}>
      <Translation passThrough="Unchecked" />
    </Checkbox>,
  );
});

let originalPlatform;
beforeEach(() => (originalPlatform = Platform.OS));
afterEach(() => (Platform.OS = originalPlatform));

const createStyle = () =>
  StyleSheet.create({
    iconStyle: {
      color: defaultTokens.paletteProductNormal,
      ios: {
        width: 26,
        height: 26,
        fontSize: 28,
      },
      android: {
        width: 24,
        height: 24,
        borderWidth: 1,
        borderColor: defaultTokens.paletteProductNormal,
        borderRadius: 100,
        fontSize: 24,
      },
    },
  });

it('works with iOS', () => {
  Platform.OS = 'ios';
  expect(createStyle()).toEqual({
    iconStyle: {
      color: defaultTokens.paletteProductNormal,
      width: 26,
      height: 26,
      fontSize: 28,
    },
  });
});

it('works with Android', () => {
  Platform.OS = 'android';
  expect(createStyle()).toEqual({
    iconStyle: {
      color: defaultTokens.paletteProductNormal,
      width: 24,
      height: 24,
      borderWidth: 1,
      borderColor: defaultTokens.paletteProductNormal,
      borderRadius: 100,
      fontSize: 24,
    },
  });
});

it('checked on ios', () => {
  Platform.OS = 'ios';

  PlaygroundRenderer.render(
    <Checkbox onPress={noop} isChecked={true}>
      <Translation passThrough="Checked" />
    </Checkbox>,
  );
});

it('checked on android', () => {
  Platform.OS = 'android';

  PlaygroundRenderer.render(
    <Checkbox onPress={noop} isChecked={true}>
      <Translation passThrough="Checked" />
    </Checkbox>,
  );
});
