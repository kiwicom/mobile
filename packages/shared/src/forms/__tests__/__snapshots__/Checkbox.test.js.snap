// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checked on android 1`] = `
<TouchableWithoutFeedback
  accessibilityComponentType="button"
  accessibilityTraits="button"
  disabled={false}
  onPress={[Function]}
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "paddingBottom": 8,
          "paddingTop": 8,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 3,
          "height": 26,
          "justifyContent": "center",
        }
      }
    >
      <Translation
        passThrough="Checked"
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "justifyContent": "center",
        }
      }
    >
      <Icon
        color="#00A991"
        fontSize={26}
        name="circle"
        style={Object {}}
      />
    </View>
  </View>
</TouchableWithoutFeedback>
`;

exports[`checked on ios 1`] = `
<TouchableWithoutFeedback
  accessibilityComponentType="button"
  accessibilityTraits="button"
  disabled={false}
  onPress={[Function]}
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "paddingBottom": 8,
          "paddingTop": 8,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 3,
          "height": 26,
          "justifyContent": "center",
        }
      }
    >
      <Translation
        passThrough="Checked"
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "justifyContent": "center",
        }
      }
    >
      <Icon
        color="#00A991"
        fontSize={26}
        name="check-circle"
        style={Object {}}
      />
    </View>
  </View>
</TouchableWithoutFeedback>
`;

exports[`renders 1`] = `
<TouchableWithoutFeedback
  accessibilityComponentType="button"
  accessibilityTraits="button"
  disabled={false}
  onPress={[Function]}
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "paddingBottom": 8,
          "paddingTop": 8,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 3,
          "height": 26,
          "justifyContent": "center",
        }
      }
    >
      <Translation
        passThrough="Unchecked"
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "borderColor": "#BAC7D5",
            "borderRadius": 100,
            "borderWidth": 1,
            "height": 24,
            "width": 24,
          }
        }
      />
    </View>
  </View>
</TouchableWithoutFeedback>
`;
